### PR TITLE
Boilerplate

### PR DESCRIPTION
**Pull request by brilam8**
file hierarchy, a few routes (POST events to a mongodb database, GET all events from the same database), connection to a mongodb database, basic functionality

### CODE REVIEW

File: src/routes/eventRouter.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+// POST a new event to the database
+routes.post('/event', (request, res) => {
+    console.log(request.body);
+    if (!request.body || request.body == {}){
+      return res.send("no request body")
+    }
+    var event = new Event({
+        title: request.body.title,
+        description: request.body.description,
```
Comment (leestanley) [inline src/routes/eventRouter.js:30]: Gonna tag @Alacau here - the design sould match whatever is sent in the POST. Not sure if all the form componens are done

File: src/database/models/eventsModel.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+//Event model consisting of the event title, start and end time, and checkIn code
+var eventSchema = new Schema(
```
Comment (leestanley) [inline src/database/models/eventsModel.js:5]: @Alacau same here

File: src/database/mongo.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const mongoose = require('mongoose');
+const config = require('../../config.json');
+
+//mongodb database: "splashes"
+const uri = "mongodb+srv://brilam8:" + config.mongoPass + "@splash-pages-eqhd7.mongodb.net/splashes?retryWrites=true&w=majority";
```
Comment (leestanley) [inline src/database/mongo.js:5]: Should this he set in a config file or .env? @garrettluu

**Comment by leestanley** [comment]
Reviewing on mobile right now - will take a look when I wake up. If the pr is finalized be sure to requesta review from me and @garrettluu

File: src/database/mongo.js (hunk 3 above)
Comment (garrettluu) [reply src/database/mongo.js:5]: Usually it would be in a .env so you can configure it easily in production when you deploy

File: package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "purple-team-backend",
+  "version": "0.0.0",
+  "private": true,
+  "main": "server.js",
+  "scripts": {
+    "start": "nodemon ./src/server.js"
+  },
+  "dependencies": {
```
Comment (garrettluu) [inline package.json:9]: Might also want to add `dotenv` as a dependency

File: src/database/models/eventsModel.js (hunk 2 above)
Comment (Alacau) [reply src/database/models/eventsModel.js:5]: event title, event date, event location, start time, end time, check-in code, background image

im not sure if we need an event description @leestanley

File: src/routes/eventRouter.js (hunk 1 above)
Comment (Alacau) [reply src/routes/eventRouter.js:30]: event title, event date, event location, start time, end time, check-in code, background image

what's missing here is event date in this format (January 10, 2020) and event location
still need to confirm with @leestanley if we need a description

Comment (brilam8) [reply src/routes/eventRouter.js:30]: yea for sure, its flexible to change the schema/model based on what we need (so i can remove description if we dont need it).

The start time/end DATE type include the date and time in format ("2020-04-25T18:00:00Z" for April 25, 2020 at 18:00 UTC time) but we can move to a different format if its easier. Haven't worked with time zones much so I'm not sure if its easy to convert UTC to local timezone or not.

Comment (leestanley) [reply src/routes/eventRouter.js:28]: @Alacau Description doesn't seem necessary for now. The front end should send it as a JavaScript date type.

**Pull request merged**